### PR TITLE
(dbt) Override marketing attribution channel

### PR DESCRIPTION
**Pull request by carolinemestre-passculture**
# DA PR

## Describe your changes

This PR was created following [this ticket](https://www.notion.so/passcultureapp/Sessions-marketing-Fix-donn-es-T1-2024-8fd77bdb6bbd47de8e1bd0d33bc83707). There was a parameter mistake in Firebase : 

Before, to track the impact of Marketing campaigns, we attributed a campaign to a user within the 24 hours that followed. 
But in Firebase, this parameter changed and was replaced by 6 months. 
The error has been corrected, but the data related to the campaigns are therefore incorrect between 20/12/23 04/04/24 included. We created this model to override Firebase's campaign data between these two dates (this override is made in `int_firebase__event` model).

Tag a reviewer if necessacy  @MatthieuRipollPass @MatthieuRipollPassCulture 

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [x] New table (non-breaking change)
- [x] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] Fields have been snake_cased
- [ ] I have checked my modifications don't break downstream models
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I have updated the dag in cases of dependencies
- [ ] My code passes CI/CD tests
- [ ] I will post on slack review channel and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql [hunk 1]
```diff
@@ -9,20 +9,32 @@
...
+    e.name,
+    e.medium,
+    e.source,
+    e.app_version,
+    CASE WHEN event_date BETWEEN '2023-12-20' AND '2024-04-04'
+        THEN a.traffic_source ELSE e.traffic_source END AS traffic_source_new,
+    CASE WHEN event_date BETWEEN '2023-12-20' AND '2024-04-04'
+        THEN a.traffic_medium ELSE e.traffic_medium END AS traffic_medium_new,
+    CASE WHEN event_date BETWEEN '2023-12-20' AND '2024-04-04'
+        THEN a.traffic_campaign ELSE e.traffic_campaign END AS traffic_campaign_new,
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql:28]: on ne veut pas écraser les champs traffic_campaign/traffic_source/traffic_medium avec la bonne donnée ?

je trouve que ça rajoute de la complexité à la requête et si on relit cette requête dans x mois, on ne comprendra plus ces règles métier (sauf si documentées qq part)

File: orchestration/dags/data_gcp_dbt/models/analytics/firebase/firebase_aggregated_offers.sql [hunk 2]
```diff
@@ -31,7 +31,7 @@ SELECT
     SUM(
         CAST(event_name = 'HasAddedOfferToFavorites' AS INT64)
     ) AS has_added_offer_to_favorites
-FROM {{ source('analytics', 'firebase_events') }}
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/analytics/firebase/firebase_aggregated_offers.sql:34]: on peut supprimer firebase_events du fichier sources.yml

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql [hunk 3]
```diff
@@ -90,8 +104,9 @@ SELECT
     CASE WHEN event_name = "screen_view" AND firebase_screen IN ("Bookings","BookingDetails") THEN 1 ELSE 0 END AS is_screen_view_bookings,
     CASE WHEN firebase_screen = "SignupConfirmationEmailSent" OR event_name = "ContinueCGU" THEN 1 ELSE 0 END AS is_signup_completed,
     CASE WHEN firebase_screen IN ("BeneficiaryRequestSent","UnderageAccountCreated","BeneficiaryAccountCreated") THEN 1 ELSE 0 END AS is_benef_request_sent,
-    CASE WHEN event_name = "login" THEN 1 ELSE 0 END AS is_login
-FROM {{ ref("int_firebase__native_event_flattened") }}
+    CASE WHEN event_name = "login" THEN 1 ELSE 0 END AS is_login,
+FROM {{ ref("int_firebase__native_event_flattened") }} AS e
+LEFT JOIN {{ ref("int_firebase__native_attribution") }} AS a ON a.user_pseudo_id = e.user_pseudo_id AND e.event_timestamp >= a.campaign_begins_at AND e.event_timestamp < a.campaign_ends_at
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql:109]: jointure plus nécessaire après le 4/4/2024 ==> on pourrait l'ajouter dans un bloc jinja {% if ds() > date_debut and ds() < date_fin %}
OU
run le modèle en one shot qui écraserait les partitions concernées et donc pas besoin de jointure

Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql:109]: Oui je suis d'accord ! Juste à voir si jamais on décide d'appliquer le modèle d'attribution à toutes les données.

**Pull request closed without merging**